### PR TITLE
docs(contributing): Updates to issue and PR management procedures

### PR DESCRIPTION
**Pull request by danabrit**
## Description

As Gatsby and the core team have evolved, so have our PR and issue management procedures. These updates to the Contributing documents are intended to align our public-facing docs with the procedures the team currently follows.

In addition, there are some other updates to things like contributor counts and team names that reflect the passage of time and further growth and development of the Gatsby Inkteam.

### CODE REVIEW

File: docs/contributing/managing-pull-requests.md [hunk 1]
```diff
@@ -171,8 +171,9 @@ Every PR opened in the repository needs to be approved before it can be merged.
 
 Typically this is:
 
-- **gatsbyjs/core** for Code
-- **gatsbyjs/learning** for Documentation
+- **gatsbyjs/themes** for certain features like Themes, Recipes, and Admin
+- **gatsbyjs/learning** for documentation
+- **gatsbyjs/core** for any other code
```
Comment (fk) [inline docs/contributing/managing-pull-requests.md:176]: Add https://github.com/orgs/gatsbyjs/teams/website?

Comment (danabrit) [reply docs/contributing/managing-pull-requests.md:176]: Maybe? I don't know if there is a workflow around that team or not.

File: docs/contributing/how-to-label-an-issue.md [hunk 2]
```diff
@@ -28,21 +30,24 @@ You can get an invite to the team by having a Pull Request merged into the Gatsb
...
 - Choose the labels that apply to that issue
+- Include any helpful comments that guide the author toward the next steps they need to take
 - That's it - sit back and relax, maybe take a few moments to enjoy the satisfaction of a job well done
 
 The rest of this document will describe how to choose the right labels for an issue.
 
 ### Find an issue that you're interested in
 
-Start with [Gatsby's issues list](https://github.com/gatsbyjs/gatsby/issues) and scroll through until you see a recent one that strikes your interest. Alternatively, you can view the [list of unlabelled issues](https://github.com/gatsbyjs/gatsby/issues?q=is%3Aopen+is%3Aissue+no%3Alabel).
+Start with [Gatsby's issues list](https://github.com/gatsbyjs/gatsby/issues), specifically the [list of untriaged issues](https://github.com/gatsbyjs/gatsby/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+triage+needed%22).
```
Comment (marcysutton) [inline docs/contributing/how-to-label-an-issue.md:50]: The retext-spell linter is failing on the word `untriaged`, which needs to be added to the dictionary: https://github.com/gatsbyjs/gatsby/blob/master/dictionary.txt

**Comment by marcysutton** [comment]
Looks like some extra commits and changes were added to this PR, it will need to get cleaned up before we can merge.

**Comment by danabrit** [comment]
Thanks, @marcysutton. My git-fu is rusty, so I've opened a [new, fresh PR](https://github.com/gatsbyjs/gatsby/pull/25618) rather than try to clean this one up.

**Pull request closed without merging**